vowpalwabbit/offset_tree_internal.cc

```diff
@@ -0,0 +1,138 @@
+#include "offset_tree_internal.h"
+#include "vw_exception.h"
+
+namespace offset_tree {
+  tree_node::tree_node(uint32_t i, uint32_t l, uint32_t r, bool lf)
+    : id(i), left_id(l), right_id(r), is_leaf(lf) {}
+
+  bool tree_node::operator==(const tree_node& rhs) const {
+    if (this == &rhs) return true;
+    return (id == rhs.id && left_id == rhs.left_id && right_id == rhs.right_id && is_leaf == rhs.is_leaf);
+  }
+
+  bool tree_node::operator!=(const tree_node& rhs) const { return !(*this == rhs); }
+
+  min_depth_binary_tree::min_depth_binary_tree()
+    : root_idx{0}, _num_leaf_nodes{0}, _initialized{false} {}
```

Comment:
consider adding these initializers in the header and leaving the constructor as default

Reply:
Done